### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by yadaiio**
Builds on top of #12 and #22.

References: https://github.com/clue/reactphp-tar/pull/20, https://github.com/clue/reactphp-redis/pull/117, https://github.com/clue/reactphp-shell/pull/20 and https://github.com/clue/reactphp-sqlite/pull/43.

As I saw the link was currently broken in line 117 as we were referring to the PHP 5.4 features, I decided to write a ticket to the maintainers of the documentation page of PHP for asking if this was intentionally, see https://github.com/php/doc-en/issues/3280. They confirmed it and gave me another link. I chose to use this as a reference here. WDYT about this @clue ?

The officially announcement of making this decision for PHP 5+ can be read here: https://www.php.net/manual/php5.php

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,7 +118,7 @@ This is a requirement because receiving multicast datagrams requires a
 [level 2 multicast conformant](https://www.tldp.org/HOWTO/Multicast-HOWTO-2.html#ss2.2)
 socket API.
 The required multicast socket options and constants have been added with
-[PHP 5.4](http://php.net/manual/en/migration54.global-constants.php) (and up).
+[PHP 5.4](https://php-legacy-docs.zend.com/manual/php5/en/migration54.new-features) (and up).
```
Comment (clue) [inline README.md]: Good catch, thanks for looking into this and for going the extra mile to report upstream! In this particular case, I think we can safely remove the entire link as it doesn't provide a lot of value anymore nowadays. If PHP decides old documentation should no longer be hosted on their end, I would rather avoid linking to the Zend version (another external dependency that may break in the future, plus it happens to be commercial and somewhat slow).

May I ask you to update/reword this to avoid the link? (Also note the same text in `Factory.php`)

Comment (yadaiio) [reply README.md]: Thank you @clue for your input here. After discussing the wording with the help of @cassifyit we changed it into the sentence: 

> The required multicast socket options and constants have been added and
> are only available to the low level socket API (ext-sockets), not
> to the newer stream based networking API.

Should be better after all. 👍

Comment (clue) [reply README.md]: Hm. I think there's still value in stating this does in fact require PHP 5.4+. The updated sentence seems to be missing this now and appears a bit clumsy imho. Perhaps you can rephrase this to improve clarity? 👍

Comment (yadaiio) [reply README.md]: Changed it back to the old sentence (without the link) to make just a small diff. @cassifyit and I changed the `(and up)` to the `+`  sign. WDYT @clue?

**Review by SimonFrings — APPROVED** [review]
Looks good to me :+1:

**Review by clue — APPROVED** [review]
@yadaiio Thanks for the update, changes LGTM! :shipit:
